src/landing/__tests__/LoginForm.test.js

```diff
@@ -0,0 +1,43 @@
+import React from 'react';
+import {
+  findByLabelText, findByText,
+} from '@testing-library/dom';
+import { render } from '@testing-library/react';
+import userEvent from '@testing-library/user-event';
+import LoginForm from '../LoginForm';
+import '@testing-library/jest-dom/extend-expect';
+
+describe('tests for login form', async () => {
+  let container;
+  let usernameInput;
+  let passwordInput;
+  let loginButton;
+
+  beforeEach(async () => {
+    container = render(<LoginForm />).container;
+    usernameInput = await findByLabelText(container, 'Username');
+    passwordInput = await findByLabelText(container, 'Password');
+    loginButton = await findByText(container, 'Log in');
+  });
+
+  it('should be displayed as error if the username field is empty', async () => {
+    expect(usernameInput.getAttribute('aria-invalid')).toBe('true');
+  });
+
+  it('should be displayed as error if the password has less than 6 characters', async () => {
+    expect(passwordInput.getAttribute('aria-invalid')).toBe('true');
+
+    userEvent.type(passwordInput, 'pass');
+    expect(passwordInput.getAttribute('aria-invalid')).toBe('true');
```

Comment:
The tests in this file fall more on the side of checking implementation details.
It would be better if they reflect more what the user sees. You could try something like this:
```js
 expect(screen.queryByText('Password must be at least 6 characters long')).toBeInTheDocument();
```

Same idea for RegisterForm.test.js.

Reply:
In these cases, the only thing the user sees is if the field is marked as invalid or not (red-underlined), that "Password must be at least 6 characters long" is always there. That's why i choose to test that way
Is there another way i should consider?

Reply:
ok, thanks for the clarification! i thought it was the error text. in this case it's ok.